lib/stores/FileStore.js

```diff
@@ -159,7 +159,7 @@ class FileStore extends DataStore {
                 log(`[FileStore] write: File is now ${offset} bytes`);
 
                 const config = this.configstore.get(file_id);
-                if (config && parseInt(config.upload_length, 10) === offset) {
+                if (config && parseInt(config.upload_length || req.headers['upload-length'], 10) === offset) {
```

Comment:
Thanks for the PR, but I'm not sure if this is the right place. In #186 we are splitting the responsibility of stores, they just handle store things and headers and request related things happen in request handlers.

Perhaps @mitjap has some thoughts on this here

Reply:
I agree. If you're splitting concerns then this can go in a separate callback, but since the event was not firing upon a live stream ending, this was the only immediate place I could find. I have applied it as a patch. Feel free to move it somewhere else.